examples/interactive_simulation.py

```diff
@@ -0,0 +1,76 @@
+import time
+
+import mujoco
+import mujoco.viewer
+import numpy as np
+
+from ambersim.utils.io_utils import load_mj_model_from_file
+
+"""Example of launching an interactive simulation with a custom controller in
+mujoco. The controller keeps running, while the user can interact with the
+system through the GUI.
+"""
+
+
+def swingup_controller(theta, theta_dot):
```

Comment:
nit: as a matter of style+clarity throughout this repo, would you mind adding type hints to function definitions from now on? For example,
```suggestion
def swingup_controller(theta: float, theta_dot: float) -> float:
```

Reply:
Yup, done! Is there an easy way to add those checks to the pre-commit linters? Might be helpful for people like me who aren't in the habit.

Reply:
It's a good question, and I'm kind of unsure the best way to proceed. `pyright` is our type checker (which DOES run in our CI), but they [don't provide official pre-commit hooks](https://github.com/microsoft/pyright/issues/3612).

Part of the reason is that for the type checking to work properly, the venv that contains the linter (which is different from the venv that you dev in) has to have all the dependencies in it for the type checking to work correctly. This can be specified in the config file, but I'm not sure how to make this work when every dev can have their own name/path for their venv.

The alternative is to tell `pre-commit` to just install all the dependencies using the `additional_dependencies` option, but then I don't know how to handle built-from-source `mujoco`, which will be merged into `main` soon.

Reply:
If it's complicated then it's probably not worth going to a bunch of trouble to set something up IMO. It'd be nice to have if it were just flipping a switch, but probably not worth pouring much time/effort into.

Reply:
I can think about it a bit more tonight, since I probably don't have enough in the tank to do anything functional. At the bare minimum, it might be possible to catch this during CI if not with a pre-commit hook by just setting `pyright` to be stricter.